Tests/DataPipeline/Support/TestUtilities.swift

```diff
@@ -76,12 +77,24 @@ extension RawEvent {
 
 // MARK: - Helper Methods
 
-func waitUntilStarted(analytics: Analytics?) {
+func waitUntilStarted(analytics: Analytics? = DataPipeline.shared.analytics) {
     guard let analytics = analytics else { return }
+
     // wait until the startup queue has emptied it's events.
     if let startupQueue = analytics.find(pluginType: StartupQueue.self) {
         while startupQueue.running != true {
             RunLoop.main.run(until: Date.distantPast)
         }
     }
 }
+
+/// Attaches and returns plugin only if it wasn't attached previously
+func attachPlugin<P: Plugin>(analytics: Analytics? = DataPipeline.shared.analytics, plugin: P) -> P {
+    guard let analytics = analytics else { return plugin }
+
+    if analytics.find(pluginType: P.self) != nil {
+        fatalError("Plugin \(P.self) is already attached")
```

Comment:
```suggestion
        throw new Error()
```

Because these utility functions are executed in tests, I suggest that we throw errors instead of call `fatalError`. 

I suggest this because when you use `fatalError`, the entire test suite will exit early and not finish if it ever executes. But if we use throw, that 1 test function will fail and the remainder of the suite continues running.

Reply:
Throwing error will force handling them and might make things more complex. Ideally, this should only execute while writing new tests when plugin is reattached mistakenly. So I think it is better to keep `fatalError` here to keep calling this method simple.